turtleflags.py

```diff
@@ -113,14 +115,37 @@ def __init__(self):
         else:
             self._read_initial_pos()
             self._init_gnome_plugins()
-            self._get_gconf_settings()
+            self._get_gio_settings()
             self._setup_gtk()
             self._build_window()
             self._run_gnome_plugins()
             self._start_gtk()
 
-    def _get_gconf_settings(self):
-        self.client = gconf.client_get_default()
+    def _get_local_settings(self, activity_root):
+        """ return an activity-specific Gio.Settings
+        """
+        # create schemas directory if missing
+        path = os.path.join(get_path(None, 'data'), 'schemas')
+        if not os.access(path, os.F_OK):
+            os.makedirs(path)
+
+        # create compiled schema file if missing
+        compiled = os.path.join(path, 'gschemas.compiled')
+        if not os.access(compiled, os.R_OK):
+            src = '%s.gschema.xml' % self._GIO_SETTINGS
+            lines = open(os.path.join(activity_root, src), 'r').readlines()
+            open(os.path.join(path, src), 'w').writelines(lines)
```

Comment:
The schema file wasn't closed after writing to it.

Reply:
Done in a016724. @chimosky Please review and merge.